### PR TITLE
Fixes bug 1132937 - add Heroku PG support with its strange superuser requirements

### PR DESCRIPTION
**Pull request by twobraids**
this is my take on the previous PR 2643 from @selenamarie.  It is a refactoring of the ```setupdb_app.py``` as well as its associated ```PostgreSQLAlchemyManager``` class.  

@selenamarie gets credit for most of the work, I get credit for beautification and integration.

### CODE REVIEW

**Comment by twobraids** [comment]
@selenamarie  @rhelmer   r?   

However, I cannot even hope this is going to work correctly in Jenkins, or in a prod install.  we're just going to have to try it.

File: socorro/external/postgresql/setupdb_app.py [hunk 1]
```diff
@@ -270,86 +309,143 @@ def create_table(element, compiler, **kw):
...
+                self.config.logger,
+                autocommit=False
+            ) as db:
+                if 'test' not in database_name and not self.config.force:
                     confirm = raw_input(
-                        'drop database %s [y/N]: ' % self.database_name)
+                        'drop database %s [y/N]: ' % database_name)
                     if not confirm == "y":
-                        logging.warn('NOT dropping table')
+                        self.config.logger.warn('NOT dropping table')
```
Comment (twobraids) [inline socorro/external/postgresql/setupdb_app.py]: that should be "database" not "table"

**Comment by rhelmer** [comment]
Code lgtm, I'd like to do some testing before merge if that's ok.

**Comment by rhelmer** [comment]
r+

**Pull request merged**